### PR TITLE
Add imported tensor attribute vector

### PR DESCRIPTION
**Pull request by toregge**
@vekterli: please review
@geirst: FYI

### CODE REVIEW

File: searchlib/src/vespa/searchlib/attribute/imported_attribute_vector_factory.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+class AttributeVector;
+class IDocumentMetaStoreContext;
+
+namespace attribute {
+
+class BitVectorSearchCache;
+class ImportedAttributeVector;
+class ReferenceAttribute;
+
+class ImportedAttributeVectorFactory {
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/attribute/imported_attribute_vector_factory.h]: Please add class comment

File: searchlib/src/tests/attribute/imported_attribute_vector/imported_attribute_vector_test.cpp [hunk 2]
```diff
@@ -534,6 +548,89 @@ TEST("onSerializeForDescendingSort() is forwarded with remapped LID to target ve
...
+            tensor2 = createDenseTensor({ {{{"x",0}}, 12}, {{{"x", 1}}, 0} });
+        } else {
+            tensor1 = createTensor({ {{{"x","1"}}, 11} }, { "x" });
+            tensor2 = createTensor({ {{{"x","0"}}, 12} }, { "x" });
+        }
+        const std::vector<ImportedAttributeFixture::LidToLidMapping<std::shared_ptr<Tensor>>> mappings =
+            {   {DocId(2), dummy_gid(3), DocId(3), tensor1 },
+                {DocId(4), dummy_gid(7), DocId(7), tensor2 } };
+        this->template reset_with_tensor_reference_mappings<TensorAttribute, std::shared_ptr<Tensor>>(
+                ValueType::from_spec(dense ? "tensor(x[2])" : "tensor(x[})"),
```
Comment (geirst) [inline searchlib/src/tests/attribute/imported_attribute_vector/imported_attribute_vector_test.cpp]: Typo: "tensor(x[})" -> "tensor(x{})"

**Review by geirst — APPROVED** [review]
Overall looks good, please consider my comments

**Review by vekterli — APPROVED** [review]
:+1: LGTM, only minor test nit

**Pull request merged**